compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2060,6 +2060,41 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                                 }
                             }
                         }
+                        // If a character was expected and the found expression is a string literal
+                        // containing a single character, perhaps the user meant to write `'c'` to
+                        // specify a character literal (issue #92479)
+                        (ty::Char, ty::Ref(_, r, _)) if r.is_str() => {
+                            if let Ok(code) = self.tcx.sess().source_map().span_to_snippet(span) {
+                                if let Some(code) =
+                                    code.strip_prefix('"').and_then(|s| s.strip_suffix('"'))
+                                {
+                                    if code.chars().nth(1).is_none() {
```

Comment:
Oh no, I think my suggestion also fires for empty string literals `""`, sorry :worried:! I think it's too late to change in this PR since it's already approved.

```suggestion
                                    if code.chars().nth(1).is_none() && !code.is_empty() {
```

Hmm that code is not that self-explanatory.

Reply:
Good point- I'll open another PR to add the 0-character check